### PR TITLE
[darwin] Header #import hygiene

### PR DESCRIPTION
**Pull request by cbracken**
This patch applies the following changes:

* Rename a stray .cc file to .mm (connection_collection.cc).
* Migrate Objective-C `#include`s to `#import` as per the style guide.
* Framework public header `#import`s in our public headers use angle-bracket (system/library) notation so they appear as such in Flutter.framework to our users.
* `#include`/`#import` order updated to reflect the style guide order: associated header, system/standard library headers, library headers, platform-specific includes.
* Include `<cstring>` where we're using `strlen`.

Bugs: https://github.com/flutter/flutter/issues/60025

### CODE REVIEW

**Comment by cbracken** [comment]
Marking as WIP since I added the `"Foo.h"` -> `<Foo.h>` at the last minute, but what we want is going to be `Flutter/Foo.h`. Will update this afternoon.

**Review by jmagman — COMMENTED** [review]
You're going to have a bad time with the common Darwin headers. Essentially the headers can't be shared since the framework names are different. I started a branch with this, I'll push it up in a bit in case there's anything useful in there for you.

**Comment by cbracken** [comment]
I may separate this out into two changes then -- the include->import migration and a separate chunk of work for flutter/flutter#60025 (unless you'd rather do that one)

**Comment by cbracken** [comment]
Alright ripped out the library header-ification, so this is back to just include->import and header include order.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -18,6 +18,11 @@
 #include "flutter/shell/common/switches.h"
 #include "flutter/shell/common/thread_host.h"
 #include "flutter/shell/platform/darwin/common/command_line.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/connection_collection.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/profiler_metrics_ios.h"
+#include "flutter/shell/platform/darwin/ios/ios_surface.h"
+#include "flutter/shell/platform/darwin/ios/platform_view_ios.h"
 #include "flutter/shell/platform/darwin/ios/rendering_api_selection.h"
 #include "flutter/shell/profiling/sampling_profiler.h"
```
Comment (stuartmorgan) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:28]: Is the separate groupings something clang-format did? IIRC for Chromium the #include and #import statements were intermixed freely, just sorted by path.

File: shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm [hunk 2]
```diff
@@ -21,6 +19,7 @@
 #include "flutter/shell/platform/darwin/common/command_line.h"
 #include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h"
 #include "flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject_Internal.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
```
Comment (stuartmorgan) [inline shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm:22]: Several of these are ObjC, unless I'm misremembering. And they certainly have ObjC names.

**Review by stuartmorgan — APPROVED** [review]
LGTM modulo comment.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm (hunk 1 above)
Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:28]: That was me; I didn't find good guidance in the style guide on that specific issue so made a guess. I can re-merge these.

File: shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm (hunk 2 above)
Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm]: Yep good catch -- done.

Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm]: Actually still a couple more -- will get those momentarily.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm (hunk 1 above)
Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:34]: Done.

File: shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm (hunk 2 above)
Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterHeadlessDartRunner.mm]: Done. And fixed a missing copyright header in one of the files.

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
> I may separate this out into two changes then -- the include->import migration and a separate chunk of work for [flutter/flutter#60025](https://github.com/flutter/flutter/issues/60025) (unless you'd rather do that one). Thought I could be sneaky and tack that on at the last moment, and failed.

I'm happy for you to take it over if the spirit moves you 😉 , it was a much larger change than I expected https://github.com/jmagman/engine/commit/de122d122d983f30c487982c9077cb5d8403a0a3

**Pull request merged**